core/api/wallet/ledger.hpp

```diff
@@ -0,0 +1,38 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include "api/types/ledger_key_info.hpp"
+#include "common/outcome.hpp"
+#include "crypto/secp256k1/secp256k1_provider.hpp"
+#include "crypto/signature/signature.hpp"
+#include "primitives/address/address.hpp"
+#include "storage/map_prefix/prefix.hpp"
+
+namespace fc::api {
+  using crypto::secp256k1::Secp256k1ProviderDefault;
+  using crypto::signature::Signature;
+  using primitives::address::Address;
+  using storage::MapPtr;
+
+  class Ledger {
+   public:
+    Ledger(const MapPtr &store);
+
+    bool Has(const Address &address) const;
+
+    outcome::result<Signature> Sign(const Address &address,
+                                    const Bytes &data) const;
+
+    outcome::result<Address> ImportKey(const LedgerKeyInfo &key_info) const;
+
+    outcome::result<Address> New() const;
+
+   private:
+    const MapPtr &store;
```

Comment:
```suggestion
    MapPtr store;
```

Reply:
Why?

Reply:
Why not create `Ledger` object once like others (eg Keystore).  Reference to shared_ptr will dangle then.

Reply:
Ledger object is created and used only inside of LedgerWallet API methods. And the current approach allows us to save some memory, because ref is always alive when Ledger object alive.
From the other hand, it is not safety if someone will use Ledger outside of LedgerWallet, for these cases I can change the code